sql/idempotent/011-vectorizer-int.sql

```diff
@@ -613,11 +615,82 @@ begin
     select pg_catalog.jsonb_extract_path_text(config, 'vectorizer_id')::int
     into strict _vectorizer_id
     ;
-    -- look up the queue table for the vectorizer
+
+    -- get the vectorizer
     select * into strict _vec
     from ai.vectorizer v
     where v.id operator(pg_catalog.=) _vectorizer_id
     ;
+
+    commit;
+    set local search_path = pg_catalog, pg_temp;
+
+    -- create a vector index if the conditions are right. otherwise proceed to vectorizer execution
+    -- hacky. a loop that can only iterate once. we can exit the loop early if it's not time to create the index
+    -- this logic *could be moved to _vectorizer_create_vector_index, but it would be nice to have function that
```

Comment:
suggestion (but not a must) I would move this code into a _vectorizer_should_create_index(vectorizer_row) returns boolean function. That avoids some of the loop weirdness and is a bit cleaner.

Reply:
yeah. i like that

Reply:
done.